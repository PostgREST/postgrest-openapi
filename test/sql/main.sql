-- Loads the config -- only needed because we're not using actual PostgREST in the tests
SELECT postgrest.pre_config();
-- Sets the config; would be called if we called callable_root() (in main)
CALL set_server_from_configuration();


-- Ensures the config is loaded
SELECT current_setting('pgrst.db_schemas', TRUE);
SELECT string_to_array(current_setting('pgrst.db_schemas', TRUE), ',');

-- Tests postgrest_openapi_spec with parameters
SELECT count(postgrest_openapi_spec(
  schemas := string_to_array(current_setting('pgrst.db_schemas', TRUE), ','),
  postgrest_version := current_setting('pgrst.version', TRUE),
  proa_version := '0.1'::text, -- TODO: needs to be updated; put into config, and have Makefile update
  document_version := 'unset'::text
));


-- shows the default server
SELECT count(jsonb_pretty(callable_root()));
