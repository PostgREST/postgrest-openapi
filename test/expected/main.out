-- Loads the config -- only needed because we're not using actual PostgREST in the tests
SELECT postgrest.pre_config();
 pre_config 
------------
 
(1 row)

-- Sets the config; would be called if we called callable_root() (in main)
CALL set_server_from_configuration();
-- Ensures the config is loaded
SELECT current_setting('pgrst.db_schemas', TRUE);
 current_setting 
-----------------
 test
(1 row)

SELECT string_to_array(current_setting('pgrst.db_schemas', TRUE), ',');
 string_to_array 
-----------------
 {test}
(1 row)

-- Tests postgrest_openapi_spec with parameters
SELECT count(postgrest_openapi_spec(
  schemas := string_to_array(current_setting('pgrst.db_schemas', TRUE), ','),
  postgrest_version := current_setting('pgrst.version', TRUE),
  proa_version := '0.1'::text, -- TODO: needs to be updated; put into config, and have Makefile update
  document_version := 'unset'::text
));
 count 
-------
     1
(1 row)

-- shows the default server
SELECT count(jsonb_pretty(callable_root()));
 count 
-------
     1
(1 row)

